tests/helpers_tests/test_validator.py

```diff
@@ -168,6 +171,61 @@ def test_is_attribute_valid(self):
     self.assertTrue(validator.is_attribute_valid('test_attribute', ""))
     self.assertTrue(validator.is_attribute_valid('test_attribute', 'test_value'))
 
+    # test if attribute value is a number, it calls is_finite_number and returns it's result
+    with mock.patch('optimizely.helpers.validator.is_finite_number',
+                    return_value=True) as is_finite:
+      self.assertTrue(validator.is_attribute_valid('test_attribute', 5))
+
+    is_finite.assert_called_once_with(5)
+
+    with mock.patch('optimizely.helpers.validator.is_finite_number',
+                    return_value=False) as is_finite:
+      self.assertFalse(validator.is_attribute_valid('test_attribute', 5.5))
+
+    is_finite.assert_called_once_with(5.5)
+
+    if PY2:
+      with mock.patch('optimizely.helpers.validator.is_finite_number',
+                      return_value=None) as is_finite:
+        self.assertIsNone(validator.is_attribute_valid('test_attribute', long(5)))
+
+      is_finite.assert_called_once_with(long(5))
+
+  def test_is_finite_number(self):
+    """ Test that it returns true if value is a number and not NAN, INF, -INF or greater than 2^53.
+        Otherwise False.
+    """
+    # test non number values
+    self.assertFalse(validator.is_finite_number('HelloWorld'))
+    self.assertFalse(validator.is_finite_number(True))
+    self.assertFalse(validator.is_finite_number(False))
+    self.assertFalse(validator.is_finite_number(None))
+    self.assertFalse(validator.is_finite_number({}))
+    self.assertFalse(validator.is_finite_number([]))
+    self.assertFalse(validator.is_finite_number(()))
+
+    # test invalid numbers
+    self.assertFalse(validator.is_finite_number(float('inf')))
+    self.assertFalse(validator.is_finite_number(float('-inf')))
+    self.assertFalse(validator.is_finite_number(float('nan')))
+    self.assertFalse(validator.is_finite_number(int(2**53) + 1))
+    self.assertFalse(validator.is_finite_number(-int(2**53) - 1))
+    self.assertFalse(validator.is_finite_number(float(2**53) + 2.0))
+    self.assertFalse(validator.is_finite_number(-float(2**53) - 2.0))
+    if PY2:
+      self.assertFalse(validator.is_finite_number(long(2**53) + 1))
+      self.assertFalse(validator.is_finite_number(-long(2**53) - 1))
+
+    # test valid numbers
+    self.assertTrue(validator.is_finite_number(0))
+    self.assertTrue(validator.is_finite_number(5))
+    self.assertTrue(validator.is_finite_number(5.5))
+    self.assertTrue(validator.is_finite_number(float(2**53) + 1.0))
```

Comment:
https://stackoverflow.com/questions/1848700/biggest-integer-that-can-be-stored-in-a-double